jerry-core/ecma/operations/ecma-arraybuffer-object.c

```diff
@@ -553,6 +555,149 @@ ecma_builtin_arraybuffer_slice (ecma_value_t this_arg, const ecma_value_t *argum
+{
+  ecma_number_t result = ecma_get_number_from_value (stored_value);
+
+  switch (op)
+  {
+    case ECMA_ATOMICS_ADD:
+    {
+      result += ecma_get_number_from_value (val);
+      break;
+    }
+    case ECMA_ATOMICS_AND:
+    {
+      ecma_value_t op_result = do_number_bitwise_logic (NUMBER_BITWISE_LOGIC_AND, stored_value, val);
+      result = ecma_get_number_from_value (op_result);
+      ecma_free_value (op_result);
+      break;
+    }
+    case ECMA_ATOMICS_EXCHANGE:
+    {
+      result = ecma_get_number_from_value (val);
+      break;
+    }
+    case ECMA_ATOMICS_OR:
+    {
+      ecma_value_t op_result = do_number_bitwise_logic (NUMBER_BITWISE_LOGIC_OR, stored_value, val);
+      result = ecma_get_number_from_value (op_result);
+      ecma_free_value (op_result);
+      break;
+    }
+    case ECMA_ATOMICS_SUBTRACT:
+    {
+      result -= ecma_get_number_from_value (val);
+      break;
+    }
+    default:
+    {
+      JERRY_ASSERT (op == ECMA_ATOMICS_XOR);
+      ecma_value_t op_result = do_number_bitwise_logic (NUMBER_BITWISE_LOGIC_XOR, stored_value, val);
+      result = ecma_get_number_from_value (op_result);
+      ecma_free_value (op_result);
+      break;
+    }
+  }
+
+  return ecma_make_number_value (result);
+} /* ecma_atomics_operation_on_number */
+
+/**
+ * Apply atomics operation on numbers.
+ *
+ * Note: ECMA_ATOMICS_COMPAREEXCHANGE is not handled here.
+ *
+ * @return ecma value
+ */
+static ecma_value_t
+ecma_atomics_operation_on_bigint (ecma_value_t stored_value, ecma_value_t val, ecma_atomics_op_t op)
+{
+  switch (op)
+  {
+    case ECMA_ATOMICS_ADD:
+    {
+      return ecma_bigint_add_sub (stored_value, val, true);
+    }
+    case ECMA_ATOMICS_AND:
+    {
+      return ecma_bigint_and (stored_value, val);
+    }
+    case ECMA_ATOMICS_EXCHANGE:
+    {
+      return ecma_copy_value (val);
+    }
+    case ECMA_ATOMICS_OR:
+    {
+      return ecma_bigint_or (stored_value, val);
+    }
+    case ECMA_ATOMICS_SUBTRACT:
+    {
+      return ecma_bigint_add_sub (stored_value, val, false);
+    }
+    default:
+    {
+      JERRY_ASSERT (op == ECMA_ATOMICS_XOR);
+
+      return ecma_bigint_xor (stored_value, val);
+    }
+  }
+
+} /* ecma_atomics_operation_on_bigint */
+
+/**
+ * ArrayBuffer get, modify, set value in buffer
+ *
+ * See also: ES12 25.1.2.13
+ *
+ * @return ecma value
+ */
+ecma_value_t
+ecma_arraybuffer_get_modify_set_value_in_buffer (
+  ecma_value_t buffer, /**< arrayBuffer */
+  uint32_t indexed_position, /**< position of the element in buffer */
+  ecma_value_t val, /**< parameter of op, number or bigint */
+  ecma_atomics_op_t op, /**< the atomics operation to execute */
+  ecma_typedarray_type_t element_type, /**< type of the elements stored in typedArray */
+  ecma_typedarray_getter_fn_t typedarray_getter_cb, /**< getter callback of typedArray */
+  ecma_typedarray_setter_fn_t typedarray_setter_cb) /**< setter callback of typedArray */
```

Comment:
Since we have plenty of params with long name i suggest to use the doxygen comment block rather.

```c
/**
 * ArrayBuffer get, modify, set value in buffer
 *
 * See also: ES12 25.1.2.13
 *
 * @param buffer: arrayBuffer
 * ....
 * @return ecma value
 */
```